WordPress/gutenberg
# Add new API to allow inserter items to be prioritised

## What?


Adds a new API to the block list settings to allow parent blocks to define a priority for blocks in the inserter for a given inner blocks.

## Why?


In https://github.com/WordPress/gutenberg/pull/48752 we added a private API to allow blocks to be prioritised. This was so we could force `Page Link` and `Custom Link` to show up as the first two results in the inserter options for the Navigation block's list view editing mode.

However, this was always envisaged as a short term way to allow the wider feature to land in WP 6.2. This PR circles back (and is an alternative to https://github.com/WordPress/gutenberg/pull/49959) with an alternative approach which looks at this problem more holistically by assuming the requirement to prioritise blocks will likely be useful to other block's in future.

## How?


Adds a new `inserterPriority` property to inner blocks which allows the consumer to pass an array of blocks and/or variations which should be prioritised in the Inserter.

It also adds a concrete example, by way of alternating the Nav block to utilise this new property to cause the `Page Link` and `Custom Link` blocks to show up first in the Inserter.

## Testing Instructions





#### Automated Testing

Run

```
npm run test:e2e packages/e2e-tests/specs/editor/plugins/inner-blocks-prioritized-inserter-blocks.test.js
```

#### Manual Testing

- Alter the `INSERTER_PRIORITY` constant in `constants.js` (see changeset) to be:

```js
export const INSERTER_PRIORITY = [
	'core/spacer',
	'core/navigation-link/page',
	'core/navigation-link',
];
```
- Rebuild with `npm run dev`
- New Post
- Add Nav block
- Open inspector controls
- Open List View tab in inspector controls
- Click inserter
- See `core/spacer` block as first item followed by the other two block variations.

### Testing Instructions for Keyboard


## Screenshots or screencast 

Co-authored-by: Dave Smith <444434+getdave@users.noreply.github.com>
